jubilant/_juju.py

```diff
@@ -100,6 +100,39 @@ def __repr__(self) -> str:
     # Keep the public methods in alphabetical order, so we don't have to think
     # about where to put each new method.
 
+    def add_credential(
+        self,
+        cloud: str,
+        *,
+        file: str | pathlib.Path | None = None,
```

Comment:
We shouldn't support interactive mode at all in this library, so let's remove this argument and always provide `--file temp.yaml` to the Juju CLI. Perhaps we could have this signature:

```python
def add_credential(
    self,
    cloud: str,
    *
    file: str | pathlib.Path | None = None,
    yaml: str | dict[str, Any] | None = None,
    region: str | None = None,
    controller: str | None = None,
) -> None:
```

But there would be two overloads and one of `file` or `yaml` must be provided. If `yaml` is a `dict[str, Any]` it's serialized to YAML first. If the `yaml` argument is provided, it's saved to a temporary file and provided to Juju as `--file=tempfile`, similar to `add_secret`.